Sources/Hedera/Transaction.swift

```diff
@@ -321,12 +327,15 @@ public class Transaction: ValidateChecksums {
 
         let `operator` = client?.operator
 
+        self.transactionId = transactionId
         self.nodeAccountIds = nodeAccountIds
         self.maxTransactionFee = maxTransactionFee
         self.`operator` = `operator`
 
         isFrozen = true
 
+        self.sources = try TransactionSources.init(transactions: try self.makeTransactionList())
```

Comment:
This still breaks transaction ID regeneration, is there a way we can add a test for that?